### PR TITLE
Future should be set to false in the default config

### PR DESCRIPTION
**Pull request by nateberkopec**
Solves #3882

### CODE REVIEW

File: lib/jekyll/configuration.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@ class Configuration < Hash
       # Filtering Content
       'show_drafts'   => nil,
       'limit_posts'   => 0,
-      'future'        => true,           # remove and make true just default
```
Comment (parkr) [inline lib/jekyll/configuration.rb]: Oh interesting... it looks like we were going to remove it anyway. Looks like #690 is where we wrote that comment. I'm cool on keeping it.

**Comment by parkr** [comment]
Cool. Just needs a test, please!

**Comment by nateberkopec** [comment]
@parkr I'm a little confused. You're asking me to test that Jekyll::Configuration::DEFAULTS['future'] == false? I can't find any similar tests that test what the DEFAULTS constant is set to. Can you give me some guidance on what kind of test you're looking for?

**Comment by parkr** [comment]
> You're asking me to test that Jekyll::Configuration::DEFAULTS['future'] == false?

Yes I am, but not directly. I would like to see a test that ensures that a post in the future is not in the output site. This can be tested in a number of ways, one of which is a cucumber feature. The test you changed, in fact, is a good place to start. Write a post that has a year of 2020 and ensure it isn't written by default and is written when `--future` is true.

**Comment by nateberkopec** [comment]
@parkr ping!

**Comment by nateberkopec** [comment]
@parkr alright actual ping. Testing that a future post isn't written by default + it is written with jekyll --build.

File: features/create_sites.feature [hunk 2]
```diff
@@ -112,16 +112,16 @@ Feature: Create sites
...
     And I have a _posts directory
     And I have the following posts:
       | title  | date       | layout | content             |
       | entry1 | 2007-12-31 | post   | content for entry1. |
-      | entry2 | 2020-01-31 | post   | content for entry2. |
+      | entry2 | 2008-01-01 | post   | content for entry2. |
     When I run jekyll build
     Then the _site directory should exist
-    And I should see "URL: /2020/01/31/entry2/" in "_site/index.html"
+    And I should see "URL: /2008/01/01/entry2/" in "_site/index.html"
```
Comment (parkr) [inline features/create_sites.feature]: Maybe just negate this? "I should not see..."

**Comment by parkr** [comment]
Merged in 559cd6c7e70f96bbda7cac3d36aee13a03206a39

**Pull request closed without merging**